src/main/scala/fi/vm/sade/hakurekisteri/integration/virta/VirtaClient.scala

```diff
@@ -196,6 +197,12 @@ class VirtaClient(
       response \ "Body" \ "OpiskelijanKaikkiTiedotResponse" \ "Virta" \ "Opiskelija" \ "Opiskeluoikeudet" \ "Opiskeluoikeus"
     opiskeluoikeudet
       .withFilter((oo: Node) => tallennettavatOpiskeluoikeustyypit.contains((oo \ "Tyyppi").text))
+      .withFilter((oo: Node) =>
+        (oo \ "Tila")
+          .filter(t => tallennettavatOpiskeluoikeustilat.contains((t \ "Koodi").text))
+          .headOption
+          .isDefined
+      )
```

Comment:
Voiskohan tän tehdä vaan suoraan näin

```suggestion
      .withFilter((oo: Node) => 
      (oo \ "Tila").exists(t => tallennettavatOpiskeluoikeustilat.contains((t \ "Koodi").text)))
```